climakitae/util/dev_logging.py

```diff
@@ -27,20 +27,33 @@
 
 import time
 import types
-import importlib
 
 # Controls the amount of indentation for library logging
 indentation_level = 0
 
 
-def _log(func):
+def _log(func: types.FunctionType) -> types.FunctionType:
     """
-    Wraps around existing functions, adding print statements upon execution and the amount of time it takes for execution. Allows function's call-stack to be viewed for all sub-functions that also have this wrapper.
+    Wraps around existing functions, adding print statements upon execution and the
+    amount of time it takes for execution. Allows function's call-stack to be viewed
+    for all sub-functions that also have this wrapper.
+
+    Parameters
+    ----------
+    func: types.FunctionType
+        The function to be wrapped.
+
+    Returns
+    -------
+    function
+        The wrapped function with logging capabilities.
     """
 
-    def _wrapper(*args, **kwargs):
-        """Wraps timer and print statement around functions if `lib_log_enabled` is True, otherwise
-        just return the function's result."""
+    def _wrapper(*args, **kwargs) -> types.FunctionType:
```

Comment:
Out of curiosity: Are there times where we would put type hints on *args and **kwargs parameters? Or is it better to leave them bare?

Reply:
A tricky question. My personal take is no. This is a situation where `*args` and `**kwargs` are so ingrained in the definitions of python that it is understood implicitly that `*args` is a general tuple and `**kwargs` is a general dictionary. There's no way to know exactly what types these arguments are going to contain since this is a wrapper that can be used on _any_ function. Because these are here to indicate that you're passing a function through this wrapper, its better to leave them bare.